cassandane/Cassandane/Cyrus/Delete.pm

```diff
@@ -910,4 +910,17 @@ sub test_no_delete_with_children
     $self->assert_str_equals('no', $talk->get_last_completion_response());
 }
 
+sub test_no_delete_non_imap
+    :min_version_3_7
+{
+    my ($self) = @_;
+
+    my $store = $self->{store};
+    my $talk = $store->get_client();
+
+    $talk->delete('INBOX.#calendars.Default');
+    $self->assert_str_equals('no', $talk->get_last_completion_response());
+    $self->assert_matches(qr/Permission denied/i, $talk->get_last_error());
+}
```

Comment:
The test could also assert that admins are still able to delete the non-email mailbox. The admin IMAPTalk instance can be obtained with `my $admintalk = $self->{adminstore}->get_client();`